packages/components/inputs/toggle-input/src/toggle-input.styles.ts

```diff
@@ -0,0 +1,231 @@
+// TODO: @redesign cleanup
+import styled from '@emotion/styled';
+import { css } from '@emotion/react';
+import {
+  designTokens,
+  type TUseThemeResult,
+} from '@commercetools-uikit/design-system';
+import type { TToggleInputProps } from './toggle-input';
+
+type NewThemeProps = {
+  isNewTheme: boolean;
+  themedTrack: ReturnType<typeof getThemedTrack>;
+  themedThumb: ReturnType<typeof getThemedThumb>;
+};
+
+type TStyledLabelProps = Pick<TToggleInputProps, 'isDisabled' | 'size'> &
+  NewThemeProps;
+type TStyledSpanProps = Pick<TToggleInputProps, 'size'> & NewThemeProps;
+
+const getThemedTrack = (themedValue: TUseThemeResult['themedValue']) =>
+  themedValue(
+    {
+      small: {
+        width: '32px',
+        height: '16px',
+      },
+      big: {
+        width: '64px',
+        height: '32px',
+      },
+    },
+    {
+      small: {
+        width: '29px',
+        height: '12px',
+      },
+      big: {
+        width: '56px',
+        height: '24px',
+      },
+    }
+  );
+
+const getThemedThumb = (themedValue: TUseThemeResult['themedValue']) =>
+  themedValue(
+    {
+      small: {
+        diameter: '13px',
+        shift: '2px',
```

Comment:
I'm not sure what `shift` means here. Is it like a margin?

Reply:
Shift (the way I thought of it) is the relative position difference of thumb against the track. It's only needed for the calculations of the final position of the thumb when on and off (to make it easier to follow the way it's calculated).
<img width="223" alt="Screenshot 2023-03-09 at 10 51 01" src="https://user-images.githubusercontent.com/49066275/223985298-567481f5-a72c-4a4e-898e-7bcbaced4096.png">
Now I think it's definitely worth explaining in a comment.